
{% extends "blog/base.html" %}

{% block content %}

<html lang="en">
<head>
	<style>body { text-align: left;
				  background-color: #d9d9d9;
				  padding-left: 40px; }
			#content {
  			width: 600px;
  			margin: auto;

  			position: relative;
  			text-align: left;}
  			pre {
  				color: blue;
  				overflow: auto;}
  	 </style>
  <meta charset="utf-8">

  <title>Chip-Seq analysis</title>
  <meta name="description" content="ChiP-Seq">
  <meta name="author" content="ScientiPic">

  <link rel="stylesheet" href="css/styles.css?v=1.0">

  <!--[if lt IE 9]>
    <script src="http://html5shiv.googlecode.com/svn/trunk/html5.js"></script>
  <![endif]-->

</head>

<body>

<div class="container-fluid">
  	<script src="js/scripts.js"></script>
  	<!--<div style="width:950px"> -->
   <h1> Chip-Seq analysis</h1>


	<p> Chip-seq experimental design: Chip-seq is performed to find the bindning site of particular protein (Transcription factors or chromatine remodeller) or chromatin modification if pull down antibodies are against chromatin marks. For
	control a non specific isotype control antibodies are ideal and also the input DNA is
	required for peak normalisation. </p>

	<p/>Once the DNA is pulled down using the target protein antibodies along with input control, its
	fractionated into fragments of 200-400 bp size for library prepration. Liberaries are
	then barcoded and sequenced. The various sequencers are available with Illumina being the
	dominant one.The sequening can be paired or single end, paired end sequening is better but costly compared to single end. <p/>

	<p>The major goal of the Analysis is to find the peaks. Peaks is the region of DNA where read
	counts are higher compared to input or surrounding reagion. Once we obtain the peaks list
	various analysis can performed.
		<ol>
			<li>Peak anotation</li>
			<li>Functional Enrichment analysis</li>
			<li>DNA motif discovery in the peaks</li>
		</ol>
	Once the raw reads are available after sequencing following workflow can be employed for analysing the
	data, to acheives the goals listed above. However the major goal is to find statistically enriched peaks</p>



	{% load static %}
    <img src="{% static "image/Chip_seq.jpg" %}"  alt="Chip seq workflow" style="width:500px;height:600px;padding-left: 50px;">
	<!-- <img src="/Users/pk/Dropbox/Django/Chip_seq_workflow/Chip_seq.jpg" alt="Chip seq workflow" style="width:500px;height:600px;padding-left: 50px;"> -->

	
	<h2>Step 0: Quality analysis </h2>

	<p>FastQC software can be used for detail please look at FASTQC website.
		Quality file can be generated with following code:</p>

	<pre>fastqc input_R1.fastq</pre>

	<p>Once you are ok with the quality you can proccede with Alignment</p>

	<h2>Step 1: Alignment using bowtie2 </h2>

	<pre>bowtie2 -p 2 -N 1 -q ~/Genomes/hg19/hg19 -1 ~/Desktop/chip_seq_proj/input_R1.fastq  -2 ~/Desktop/chip_seq_proj/input_R2.fastq |samtools view -bS - > ~/Desktop/chip_seq_proj/bamfiles/input.bam</pre>

	<h2>Step 2: Sorting, removing duplicates, and indexing the bam file </h2>

	<p> Before we feed the aligned bam file for peak finding using MACS, we need to sort it, and remove PCR duplicates and index the bam files
	these steps can be performed with samtools commands as demonstrated below</p>

	<pre>samtools sort ~/Desktop/chip_seq_proj/bamfiles/file.bam ~/Desktop/chip_seq_proj/sorted_files/sort_file </pre>
	<pre>samtools rmdup ~/Desktop/chip_seq_proj/sorted_files/sort_file.bam ~/Desktop/chip_seq_proj/rmdup_index_files/rmdup_sort_file.bam </pre>
	<pre>samtools index ~/Desktop/chip_seq_proj/rmdup_index_files/rmdup_sort_file.bam </pre>



	<h2> Step 3: Peak Calling </h2>

	<p>Once the alignement,sorting, remove pcr duplicate and file index is done we are ready for peak calling</p>

	<pre>macs2 callpeak -t ~/Desktop/chip_seq_proj/rmdup_index_files/test.bam -c ~/Desktop/chip_seq_proj/rmdup_index_files/input.bam -f BAMPE -g hs --outdir ~/Desktop/chip_seq_proj/MACS_results -n test_vs_inp -B -q 0.05</pre>

	<p>Once the peaks are obtained The peaks can be anotated for genes using the Homer
	annotation toolslooks for the file with .narrowPeaks generated by MACS2</p>

	<h2> Step 4: Peak Annotation </h2>
	<p>Peaks obtained has coordinates of geneome, which need to be converted into usuful biological information,
	so one most commanly used information for the peaks are the nearest associated genes to peaks which can be obtained using Homer tools using the command shown below</p>

	<pre>annotatePeaks.pl ~/folderName/test_vs_inp_peaks.narrowPeak hg19 > Anotations_test_vs_Inp.txt</pre>

	<div/>
	
	<div>
	<p> The code for above described workflow can be generated from <a href="{% url 'GeneExp_view' %}">ScriptWriter</a> app on the this site.</p>
	 </div>
	  
	 
</div>
</body>

{% endblock %}
